doc/error-catalog.md

```diff
@@ -187,6 +187,27 @@ test/runtime-errors.sh test-external_cmd_typed_args
 - Did you forget to source a file that contains the proc or shell function you
   wanted to run?
 
+### OILS-ERR-201
+
+<!--
+Generated with:
+test/runtime-errors.sh test-arith_ops_str
+-->
+
+```
+  = "100" + "10a"
+          ^
+[ -c flag ]:1: fatal: Binary operator expected numbers, got Str and Str (OILS-ERR-201)
+```
+
+- The `+`, `-`, `*` and `/` operators are intended to represent their
```

Comment:
I would start this with more "actionable" / direct advice  questions, particularly:

> Did you mean to use `++` to concatenate strings? 

> `+` is for addition on numbers, while `++` is for concenation.

I think that is going to be the 99% case

Also might be reasonable to try to use something "realistic" like `= "spam" + myvar` or something

Reply:
> Also might be reasonable to try to use something "realistic" like `= "spam" + myvar` or something

Hmm, I want to make sure that the example makes the error clear. I worry that `= "spam" + myvar` doesn't make the issue obvious enough.

Reply:
I decided to include two examples of the error, the Str + Str case (where concatenation should be used) and the Str + Int case (where the Str cannot be coerced).